Unity-C#/README.md

```diff
@@ -0,0 +1,252 @@
+In C# where the braces are placed is up to the user, however the C# MSDN standard is on a new line.  That said as long as the project has agreed to use on the same line there should be no complaints.
+
+
+## Naming
+All names SHOULD be written in English, American English.
+### In general:
+##### Variable names
+The reason that underscore use of private variables is favoured even though it is generally not standard is because Properties are not supported in Unity so the standard public accessors notation for C# and the MSDN guidelines can not be used effectively.
+```csharp
+public int aName;
+private int _aName;
+```
+
+
+##### Property names
+In Unity properties are not supported in the editor, and their use is discouraged.  If you want to have a member in the Unity Editor you need to use a public member (which is against MSDN coding guidelines).
+```csharp
+public int Name { get; set; }
+```
+
+
+##### Method names
+
+```csharp
+public void Rename(int name)
+```
+
+
+##### Class names
+```csharp
+public class User
+```
+
+
+##### Constants
+```csharp
+public const int MyNumber = 42;
+```
+
+
+##### Enum
+```csharp
+public enum Name
+```
+
+
+##### Specific Naming Conventions
+
+- The "is" prefix SHOULD be used for boolean variables and methods. Alternatives include "has", "can" and "should".
+
+- The terms "initialize" or "init" CAN be used where an object or a concept is established.
+
+- Plural form MUST be used to name collections.
+
+- Iterator variables SHOULD be called "i", "j", "k", etc.
+
+- Complement names MUST be used for complement entities. Examples: get/set, add/remove, create/destroy, start/stop, insert/delete, begin/end, etc.
+
+- Negated boolean variable names MUST be avoided:
+```csharp
+	bool isNotError, isNotFound; // WRONG.
+```
+
+- Methods returning an object MAY be named after what they return, and methods returning void after what they do.
+
+
+## Variable declarations
+Declare one variable per type statement, it makes it easier to re-order the lines, and put declarations in the order in which they make sense.
+Ideally, variables should be initialized where they are declared and they must be declared in the smallest scope possible.
+
+
+### Loop
+Only loop control statements MUST be included in the "for" loop construction.
+
+Loop variables should be initialized immediately before the loop; loop variables in a "for" statement may be initialized in the "for" loop construction.
+
+The use of "break" and "continue" is not discouraged.
+
+### Conditionals
+Complex conditional expressions SHOULD be avoided; use temporary boolean variables instead with proper naming.
+
+The nominal case should be put in the "if" part and the exception in the "else" part of an "if" statement.
+
+Executable statements in conditionals MUST be avoided.
+
+### Miscellaneous
+The use of magic numbers in the code should be avoided; they should be declared using named **constants** instead.
+
+
+## Object and Array creation
+```csharp
+
```

Comment:
Not sure its required, I was thinking about putting in array instantiation and the use of var key word, but I'm not sure it is necessary.